test/mysql_tests.erl

```diff
@@ -824,18 +830,38 @@ decimal_trunc(_Pid) ->
     ?assertMatch("Note 1265: Data truncated for column 'balance'" ++ _,
                  LoggedWarning2),
     %% Decimal sent as DECIMAL => no warning
-    {ok, ok, []} = error_logger_acc:capture(fun () ->
-        ok = mysql:execute(Pid, decr, [{decimal, <<"10.2">>}, 3]),
-        ok = mysql:execute(Pid, decr, [{decimal, "10.2"}, 3]),
-        ok = mysql:execute(Pid, decr, [{decimal, 10.2}, 3]),
-        ok = mysql:execute(Pid, decr, [{decimal, 10.2}, 3]),
-        ok = mysql:execute(Pid, decr, [{decimal, 0}, 3]) % <- integer coverage
-    end),
+     {ok, ok, MaybeWarning} = error_logger_acc:capture(fun () ->
+         ok = mysql:execute(Pid, decr, [{decimal, <<"10.2">>}, 3]),
+         ok = mysql:execute(Pid, decr, [{decimal, "10.2"}, 3]),
+         ok = mysql:execute(Pid, decr, [{decimal, 10.2}, 3]),
+         ok = mysql:execute(Pid, decr, [{decimal, 10.2}, 3]),
+         ok = mysql:execute(Pid, decr, [{decimal, 0}, 3]) % <- integer coverage
+     end),
     ?assertMatch({ok, _, [[1, 4959.2], [2, 4959.2], [3, 4959.2]]},
                  mysql:query(Pid, <<"SELECT id, balance FROM test_decimals">>)),
+    assert_decimal_trunctation_warning(MaybeWarning),
     ok = mysql:query(Pid, "DROP TABLE test_decimals"),
     ok = mysql:stop(Pid).
 
+%% Asserts when running CI
+%% Assume MYSQL_IMAGE is set to 'mariadb' or 'mysql',
+%% and MYSQL_VERSION is also set respectively
+assert_decimal_trunctation_warning([]) -> ok;
+assert_decimal_trunctation_warning([{warning_msg, Msg}]) ->
+    ?assertMatch("Note 1265: Data truncated for column 'balance'" ++ _, Msg),
+    case os:getenv("MYSQL_IMAGE") of
+        "mariadb" ->
+            throw("not expecting mariadb to emit warning " ++ Msg);
+        _ ->
+            ok
+    end,
+    case {os:getenv("MYSQL_IMAGE"), os:getenv("MYSQL_VERSION")} of
```

Comment:
Let's avoid depending on external variables. There are already mechanisms in this test suite to ask the server for its version: `db_version_string(Pid)`, `parse_db_version(Version)` and `is_mariadb(Version)`.

Look at the conditional code for the JSON test for inspiration:

```erlang
json(Pid) ->
    Version = db_version_string(Pid),
    try
        is_mariadb(Version) andalso throw(no_mariadb),
        Version1 = parse_db_version(Version),
        Version1 >= [5, 7, 8] orelse throw(version_too_small)
    of _ ->
        test_valid_json(Pid),
        test_invalid_json(Pid)
    catch
        throw:no_mariadb ->
            error_logger:info_msg("Skipping JSON test, not supported on"
                                  " MariaDB.~n");
        throw:version_too_small ->
            error_logger:info_msg("Skipping JSON test. Current MySQL version"
                                  " is ~s. Required version is >= 5.7.8.~n",
                                  [Version])
    end.
```